File: Java/Easy/Array/RemoveDuplicate.java
/*
nums = [1,1,2],
output = 2
nums = [0,0,1,1,1,2,2,3,3,4],
output = 5
i and j 

1,1,2
1,2 1

0,1,2,3,4,0,2,1,3,1


*/

class Solution {
    public int removeDuplicates(int[] nums) {
    	if(nums.length==0|| nums.length==1)
    	{
    		return nums.length;
    	}
        int i=1;
        int j =1;
        int prev = nums[0];//0

        while(j<nums.length)
        {
        	if(nums[j]!=prev)//4
        	{
        		prev = nums[j];//4
        		swap(nums,i,j);
        		i++;
        	}
        	j++;
        }
        return i;
    }
    private void swap(int[] nums, int i, int j)
    {
    	int temp = nums[i];
    	nums[i] = nums[j];
    	nums[j] = temp;
    }
}

File: CSharp/Easy/Array/RemoveDuplicate.cs
/*
Example 1:

Given nums = [1,1,2],

Your function should return length = 2, with the first two elements of nums being 1 and 2 respectively.

It doesn't matter what you leave beyond the returned length.
Example 2:

Given nums = [0,0,1,1,1,2,2,3,3,4],

Your function should return length = 5, with the first five elements of nums being modified to 0, 1, 2, 3, and 4 respectively.

It doesn't matter what values are set beyond the returned length.
[0,0,1,1,1,2,2,3,3,4]

0,1,2,3,4,0,2,1,3,4
*/
public class Solution {
    public int RemoveDuplicates(int[] nums) {
    	if(nums.Length==0 || nums.Length==1)
    	{
    		return nums.Length;
    	}
        int slow = 0;
        int fast = 1;
        while(fast<nums.Length)
        {
        	if(nums[fast]!=nums[slow])
        	{
        		slow++;
        		nums[slow] = nums[fast];
        	}
        	fast++;
        }
        return slow+1;
    }
}